src/backend/looped/weapons/aimbot.cpp

```diff
@@ -62,9 +63,38 @@ namespace big
 
 			if (PAD::GET_DISABLED_CONTROL_NORMAL(0, (int)ControllerInputs::INPUT_AIM))
 			{
-				Vector3 camera_target = aim_lock - CAM::GET_GAMEPLAY_CAM_COORD();
-				float camera_heading  = atan2f(camera_target.x, camera_target.y) * 180.0 / 3.14159265358979323846;
-				float magnitude       = sqrtf(camera_target.x * camera_target.x + camera_target.y * camera_target.y
+				Vector3 camera_target;
+
+				if (g.weapons.aimbot.smoothing)
+				{
+					//Avoid buggy cam
+					if (using_aimbot_first_time) {
+						Vector3 cam_coords    = CAM::GET_GAMEPLAY_CAM_COORD();
+						Vector3 cam_rot       = CAM::GET_GAMEPLAY_CAM_ROT(0);
+						Vector3 cam_direction = math::rotation_to_direction(cam_rot);
+						float distance        = 150;
+						Vector3 multiply      = cam_direction * distance;
+						Vector3 front_cam     = cam_coords + multiply;
+						camera_target         = front_cam - CAM::GET_GAMEPLAY_CAM_COORD();
+						smooth_factor           = camera_target;
+						using_aimbot_first_time = false;
+					}
+					Vector3 target = aim_lock - CAM::GET_GAMEPLAY_CAM_COORD();
+					smooth_factor.x += (target.x - smooth_factor.x) * g.weapons.aimbot.smoothing_speed / 10.f;
+					smooth_factor.y += (target.y - smooth_factor.y) * g.weapons.aimbot.smoothing_speed / 10.f;
+					smooth_factor.z += (target.z - smooth_factor.z) * g.weapons.aimbot.smoothing_speed / 10.f;
+
+					camera_target = smooth_factor;
+				}
+				else
+				{
+					camera_target = aim_lock - CAM::GET_GAMEPLAY_CAM_COORD();
+				}
+				if (aim_lock.x == 0.f && aim_lock.y == 0.f && aim_lock.z == 0.f)
+					return;
+
+				float camera_heading = atan2f(camera_target.x, camera_target.y) * 180.0 / 3.14159265358979323846;
```

Comment:
You are calling atan2f which will return an imprecise float, but then you are using 3.14159265358979323846 which is a double as you did not tell the compiler to truncate it to a float with f and are declaring it over the scale of a float. The compiler will implicitly do this:

Calculate atan2. Cast it a float from a double. Multiply the result by 180.0. Cast it to a double. Divide by 3.14159265358979323846. Cast it back to a float. You must remember that the compiler follows PEMDAS, and that a casting operating counts as a parenthesis, it always takes precedence. If this was some shit little code for the menu, I couldn't even be assed to write this, I would let Yim deal with this, but since this running every frame, up to 300 times per second, the code has to be immaculately efficient. 

Instead, you want to use something like M_PI.